cmd/plugin-manager/add/add.go

```diff
@@ -0,0 +1,196 @@
+	cmd.Flags().StringVarP(&o.Version, "version", "", "", "Install specific plugin version (if not passed, installs latest plugin version or the only available one)")
+}
+
+func (o *Options) run(args []string) error {
+	log := o.globalFlags.GetLogger()
+
+	manifestMap, err := plugin.BuildManifestMap(log, args[0], o.Repo)
+	if err != nil {
+		return nil
+	}
+
+	path := BuildInstallBinaryPath(o.PluginDir)
+	installVersion := "latest"
+	if o.Version != "" {
+		installVersion = o.Version
+	}
+
+	if len(manifestMap) > 1 {
+		// if the plugin is found across multiple repository then fail and ask for a specific repo
+		log.Errorf(fmt.Sprintf("The plugin %s is found across multiple repos, please specify one repo with --repo flag", args[0]))
+	} else if len(manifestMap) == 1 {
+		// the plugin is found in only one repo
+		for repo, manifest := range manifestMap {
+			// install the only available version of the plugin
+			if len(manifest) == 1 {
+				for _, value := range manifest {
+					// check if the version is mentioned and matches the version in manifest file
+					if value.Name != "" && (o.Version == "" || string(value.Version) == o.Version) {
+						err := DownloadBinary(fmt.Sprintf("%s/%s", path, repo), value.Name, value.Binaries[0].URI)
+						if err != nil {
+							return err
+						}
+						fmt.Println(fmt.Sprintf("Adding plugin %s from repo %s", args[0], repo))
+						log.Infof("plugin %s added to the path - %s", args[0], fmt.Sprintf("%s/%s", path, repo))
+						return nil
+					} else {
+						log.Errorf(fmt.Sprintf("The version %s of plugin %s is not available", o.Version, args[0]))
+					}
+				}
+			} else if len(manifest) > 1 {
+				// if there are multiple version of the plugins are available then look for the latest or mentioned version and if not found fail and ask user to input a version using --version flag
+				for repo, manifest := range manifestMap {
+					for _, value := range manifest {
+						if string(value.Version) == installVersion {
+							err := DownloadBinary(fmt.Sprintf("%s/%s", path, repo), value.Name, value.Binaries[0].URI)
+							if err != nil {
+								return err
+							}
+							fmt.Println(fmt.Sprintf("Adding plugin %s from repo %s", args[0], repo))
+							log.Infof("plugin %s added to the path - %s", args[0], fmt.Sprintf("%s/%s", path, repo))
+							return nil
+						}
+					}
+					log.Errorf(fmt.Sprintf("The %s version of the plugin %s is not found, please mention one version using --version", installVersion, args[0]))
+				}
+			} else {
+				// throw error saying that the plugin doest exists
+				log.Errorf(fmt.Sprintf("The plugin %s is not found", args[0]))
+			}
+		}
+	} else {
+		// throw error saying that the plugin doest exists
+		log.Errorf(fmt.Sprintf("The plugin %s is not found", args[0]))
+	}
+	return nil
+}
+
+func DownloadBinary(filepath string, filename string, url string) error {
```

Comment:
Should this be exported?

Reply:
this function is not used anywhere else, I can make it not exposed in the follow up PR.